src/authn/ccloudProvider.ts

```diff
@@ -138,14 +137,14 @@ export class ConfluentCloudAuthProvider implements vscode.AuthenticationProvider
     if (authCallback === undefined) {
       // user cancelled the operation
       logger.debug("createSession() user cancelled the operation");
-      return Promise.reject(new Error("User cancelled the authentication flow."));
+      throw new Error("User cancelled the authentication flow.");
     }
 
     if (authCallback.resetPassword) {
       // user reset their password, so we need to notify them to reauthenticate
       logger.debug("createSession() user reset their password");
       this.showResetPasswordNotification();
-      return Promise.reject(new Error("User reset their password."));
+      throw new Error("User reset their password.");
```

Comment:
Other parts of this method are throwing errors, and we aren't worried about them going to Sentry after the client [init overhaul](https://github.com/confluentinc/vscode/pull/1311)